docs/index.md

```diff
@@ -1225,7 +1225,7 @@ Hooks associated to `presync` events are triggered before each release is instal
 This is the ideal event to execute any commands that may mutate the cluster state as it will not be run for read-only operations like `lint`, `diff` or `template`.
 
 `preapply` hooks are triggered before a release is uninstalled, installed, or upgraded as part of `helmfile apply`.
-This is the ideal event to hook into when you are going to use `helmfile apply` for every kind of change, and you want the hook to be called only when any kind of change is being made.
+This is the ideal event to hook into when you are going to use `helmfile apply` for every kind, Regardless of whether the releases have changed, `preapply` will be executed, please ensure that `preapply` executes the command is idempotent.
```

Comment:
```suggestion
This is the ideal event to hook into when you are going to use `helmfile apply` for every kind of change and you want the hook to be triggered regardless of whether the releases have changed or not. Be sure to make each `preapply` hook command idempotent. Otherwise, rerunning helmfile-apply on a transient failure may end up either breaking your cluster, or the hook that runs for the second time will never succeed.
```